core/common/src/YearMonth.kt

```diff
@@ -0,0 +1,396 @@
+ *
+ * @sample kotlinx.datetime.test.samples.YearMonthSamples.yearMonth
+ */
+public val LocalDate.yearMonth: YearMonth get() = YearMonth(year, month)
+
+/**
+ * Combines this year-month with the specified day-of-month into a [LocalDate] value.
+ *
+ * @sample kotlinx.datetime.test.samples.YearMonthSamples.onDay
+ */
+public fun YearMonth.onDay(day: Int): LocalDate = LocalDate(year, month, day)
+
+/**
+ * Returns the number of whole years between two year-months.
+ *
+ * The value is rounded toward zero.
+ *
+ * @see YearMonth.until
+ * @sample kotlinx.datetime.test.samples.YearMonthSamples.yearsUntil
+ */
+public fun YearMonth.yearsUntil(other: YearMonth): Int =
+    ((other.prolepticMonth - prolepticMonth) / 12L).toInt()
+
+/**
+ * Returns the number of whole months between two year-months.
+ *
+ * If the result does not fit in [Int], returns [Int.MAX_VALUE] for a positive result
+ * or [Int.MIN_VALUE] for a negative result.
+ *
+ * @see YearMonth.until
+ * @sample kotlinx.datetime.test.samples.YearMonthSamples.monthsUntil
+ */
+public fun YearMonth.monthsUntil(other: YearMonth): Int =
+    (other.prolepticMonth - prolepticMonth).clampToInt()
+
+/**
+ * Returns the whole number of the specified month-based [units][unit] between `this` and [other] year-months.
+ *
+ * The value returned is:
+ * - Positive or zero if this year-month is earlier than the other.
+ * - Negative or zero if this year-month is later than the other.
+ * - Zero if this date is equal to the other.
+ *
+ * The value is rounded toward zero.
+ *
+ * @see YearMonth.monthsUntil
+ * @see YearMonth.yearsUntil
+ * @sample kotlinx.datetime.test.samples.YearMonthSamples.until
+ */
+public fun YearMonth.until(other: YearMonth, unit: DateTimeUnit.MonthBased): Long =
+    (other.prolepticMonth - prolepticMonth) / unit.months
+
+/**
+ * The [YearMonth] 12 months later.
+ *
+ * @throws DateTimeArithmeticException if the result exceeds the boundaries of [YearMonth].
+ * @sample kotlinx.datetime.test.samples.YearMonthSamples.plusYear
+ */
+public fun YearMonth.plusYear(): YearMonth = plus(1, DateTimeUnit.YEAR)
+
+/**
+ * The [YearMonth] 12 months earlier.
+ *
+ * @throws DateTimeArithmeticException if the result exceeds the boundaries of [YearMonth].
+ * @sample kotlinx.datetime.test.samples.YearMonthSamples.minusYear
+ */
+public fun YearMonth.minusYear(): YearMonth = minus(1, DateTimeUnit.YEAR)
+
+/**
+ * The [YearMonth] one month later.
+ *
+ * @throws DateTimeArithmeticException if the result exceeds the boundaries of [YearMonth].
+ * @sample kotlinx.datetime.test.samples.YearMonthSamples.plusMonth
+ */
+public fun YearMonth.plusMonth(): YearMonth = plus(1, DateTimeUnit.MONTH)
+
+/**
+ * The [YearMonth] one month earlier.
+ *
+ * @throws DateTimeArithmeticException if the result exceeds the boundaries of [YearMonth].
+ * @sample kotlinx.datetime.test.samples.YearMonthSamples.minusMonth
+ */
+public fun YearMonth.minusMonth(): YearMonth = minus(1, DateTimeUnit.MONTH)
```

Comment:
These 4 functions should be plural and accept an int parameter that defaults to 1 as in:
```
public fun YearMonth.minusMonths(months: Int = 1): YearMonth = minus(months, DateTimeUnit.MONTH)
```

Reply:
In our research, we haven't found cases where the value was anything other than `1`, so the extra flexibility is pointless. You have a more general `plus` for this case if `1` doesn't suit your needs.